### ISSUE TITLE
MissingGlyphId error while compiling GS after #549

### ISSUE THREAD

**Issue by anthrotype** [googlefonts/fontc#555]
this is possibly a regression from https://github.com/googlefonts/fontc/pull/549

"K.alt" is among the glyphs in the public.skipExportGlyphs set. It doesn't look like "K.alt" is mentioned in the hand-written features.fea (I could not grep it), so we must be incorrectly introducing it when autogenerating either kerning or mark/mkmk

```
clupo@clupo-macbookpro3 ~/oss/googlesans (main)
$ RUST_BACKTRACE=1 fontc -s source/GoogleSans/GoogleSans.designspace
thread '<unnamed>' panicked at fontbe/src/features.rs:525:33:
called `Result::unwrap()` on an `Err` value: MissingGlyphId(K.alt)
stack backtrace:
   0: rust_begin_unwind
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs:595:5
   1: core::panicking::panic_fmt
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/panicking.rs:67:14
   2: core::result::unwrap_failed
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/result.rs:1652:5
   3: <fontbe::features::FeatureWriter as fea_rs::compile::feature_writer::FeatureProvider>::add_features
   4: fea_rs::compile::compile_ctx::CompilationCtx::compile
   5: fea_rs::compile::compiler::Compiler::compile
   6: <fontbe::features::FeatureWork as fontdrasil::orchestration::Work<fontbe::orchestration::Context,fontbe::orchestration::AnyWorkId,fontbe::error::Error>>::exec
   7: fontc::work::AnyWork::exec
   8: std::panicking::try
   9: <rayon_core::job::HeapJob<BODY> as rayon_core::job::Job>::execute
  10: rayon_core::registry::WorkerThread::wait_until_cold
  11: rayon_core::registry::ThreadBuilder::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
[2023-11-09T09:47:49.742882Z main fontc::workload ERROR] Be(Features) failed Job panicked: 'called `Result::unwrap()` on an `Err` value: MissingGlyphId(K.alt)'
At least one work item failed
```

**Comment by anthrotype** [comment]
I think this occurs because "K.alt" is mentioned in the groups.plist (used for kerning classes) and our feature writer expect all glyphs listed in those groups to be present in the font and raises an error if they are missing; it should just skip them if not present in the new glyph order (after skipped glyphs have been pruned).

**Comment by anthrotype** [comment]
ufo2ft's kernFeatureWriter skips non-export glyphs when parsing the groups, or the entire group if emptied:

https://github.com/googlefonts/ufo2ft/blob/efb4658890ba645292a9c6bc055c91afe225c4a1/Lib/ufo2ft/featureWriters/kernFeatureWriter.py#L309-L311

similarly it skips the entire kerning pair if any of the glyphs is missing

https://github.com/googlefonts/ufo2ft/blob/efb4658890ba645292a9c6bc055c91afe225c4a1/Lib/ufo2ft/featureWriters/kernFeatureWriter.py#L340-L344

the source format's kerning and groups data can and do contain glyphs that may not be in the font and we should handle this gracefully by simply skipping those

**Comment by rsheeter** [comment]
Good catch, nice to have a concrete example. I generally haven't updated
anything but glyph order to obey skip export yet.

**Comment by anthrotype** [comment]
Further clarification. When collecting kerning pairs into our IRs, we do already prune pairs when any of the single glyph participants do not exist in the glyph order. However when collecting the glyph groups, we let non-export glyphs in the kerning groups IR, but the the fontbe kern feature generator arbitrarily assigns gid 0 (.notdef?!) to missing glyphs when construcing the coverages which is wrong. Since we are pruning pairs from the kerning IR when single glyphs are not in the glyph order, we should similarly also prune missing/non-export glyphs when they occur inside a kerning group as we construct the kerning IR.

**Comment by anthrotype** [comment]
I'm the non-export glyph inside kerning group issue in a follow up PR, hold on

**Comment by anthrotype** [comment]
Fixed by #558
